### PR TITLE
Updates VectorInvariant advection scheme for curvilinear grids

### PR DESCRIPTION
**Pull request by glwagner**
This PR changes the relative vorticity term and Bernoulli term gradient in the `VectorInvariant` advection scheme to be valid on curvilinear grids.

Follows: https://mitgcm.readthedocs.io/en/latest/algorithm/algorithm.html#vector-invariant-momentum-equations

### CODE REVIEW

File: src/Models/HydrostaticFreeSurfaceModels/hydrostatic_free_surface_advection.jl [hunk 1]
```diff
@@ -1,25 +1,29 @@
...
+@inbounds ζ₂wᶠᶜᶠ(i, j, k, grid, u, w) = ℑxᶠᵃᵃ(i, j, k, grid, w) * δzᵃᵃᶠ(i, j, k, grid, u) / Δzᵃᵃᶠ(i, j, k, grid)
+@inbounds ζ₁wᶜᶠᶠ(i, j, k, grid, v, w) = ℑyᵃᶠᵃ(i, j, k, grid, w) * δzᵃᵃᶠ(i, j, k, grid, v) / Δzᵃᵃᶠ(i, j, k, grid)
 
 @inline U_dot_∇u(i, j, k, grid, advection::VectorInvariant, U) = (
-    - ℑyᵃᶜᵃ(i, j, k, grid, ζ₃ᶠᶠᵃ, U.u, U.v) * ℑxyᶠᶜᵃ(i, j, k, grid, U.v)
-    + δxᶠᵃᵃ(i, j, k, grid, Khᶜᶜᶜ, U.u, U.v) / Δx(i, j, k, grid)
-    + ℑzᵃᵃᶜ(i, j, k, grid, ζ₂wᶠᶜᶠ, U.u, U.w))
+    - ℑyᵃᶜᵃ(i, j, k, grid, ζ₃ᶠᶠᵃ, U.u, U.v) * ℑxᶠᵃᵃ(i, j, k, grid, ℑyᵃᶜᵃ, Δx_vᶜᶠᵃ, U.v) / Δxᶠᶜᵃ(i, j, k, grid) # Vertical relative vorticity term
+    + ∂xᶠᶜᵃ(i, j, k, grid, Khᶜᶜᶜ, U.u, U.v)    # Bernoulli head term
+    + ℑzᵃᵃᶜ(i, j, k, grid, ζ₂wᶠᶜᶠ, U.u, U.w))  # Horizontal vorticith / vertical advection term
```
Comment (francispoulin) [inline src/Models/HydrostaticFreeSurfaceModels/hydrostatic_free_surface_advection.jl:21]: typo in the comment: vorticith -> vorticity

File: src/Models/HydrostaticFreeSurfaceModels/hydrostatic_free_surface_advection.jl [hunk 2]
```diff
@@ -1,25 +1,29 @@
...
 
 @inline U_dot_∇v(i, j, k, grid, advection::VectorInvariant, U) = (
-    + ℑxᶜᵃᵃ(i, j, k, grid, ζ₃ᶠᶠᵃ, U.u, U.v) * ℑxyᶜᶠᵃ(i, j, k, grid, U.u)
-    + δyᵃᶠᵃ(i, j, k, grid, Khᶜᶜᶜ, U.u, U.v) / Δy(i, j, k, grid)
-    + ℑzᵃᵃᶜ(i, j, k, grid, ζ₁wᶜᶠᶠ, U.v, U.w))
+    + ℑxᶜᵃᵃ(i, j, k, grid, ζ₃ᶠᶠᵃ, U.u, U.v) * ℑyᵃᶠᵃ(i, j, k, grid, ℑxᶜᵃᵃ, Δy_uᶠᶜᵃ, U.u) / Δyᶜᶠᵃ(i, j, k, grid) # Vertical relative vorticity term
+    + ∂yᶜᶠᵃ(i, j, k, grid, Khᶜᶜᶜ, U.u, U.v)   # Bernoulli head term
+    + ℑzᵃᵃᶜ(i, j, k, grid, ζ₁wᶜᶠᶠ, U.v, U.w)) # Horizontal vorticith / vertical advection term
```
Comment (francispoulin) [inline src/Models/HydrostaticFreeSurfaceModels/hydrostatic_free_surface_advection.jl:26]: Same typo in comment

**Review by francispoulin — APPROVED** [review]
Besides minor typos in the comments, which I don't suppose you plan to keep, the changes look good to me.

File: src/Models/HydrostaticFreeSurfaceModels/hydrostatic_free_surface_advection.jl (hunk 1 above)
Comment (glwagner) [reply src/Models/HydrostaticFreeSurfaceModels/hydrostatic_free_surface_advection.jl]: I think we'll keep them as long as we need to! More comments is usually better I think.

Comment (francispoulin) [reply src/Models/HydrostaticFreeSurfaceModels/hydrostatic_free_surface_advection.jl]: Okay, then probably good to correct the typos.

**Pull request merged**